### PR TITLE
Refactor to templ/datasetediting suggest publications (and dataset/show_publications)

### LINKED ISSUE #1486: [templ] Convert views for datasetediting/publication.go to templ
Includes:

* `views/dataset/_suggest_publications.gohtml`
* `views/dataset/add_publication.gohtml`
*  `views/dataset/refresh_publications.gohtml`
* `views/dataset/show_publications.gohtml`

### PR DESCRIPTION
**Pull request by nicolasfranck**
fixes #1486

also done:

* adds `views/dataset/sub_nav.templ` as was done previously in `views/publication/sub_nav.templ`
* fix bug in `views/publication/summary.templ` (double footer, or nulpointer exception)

### CODE REVIEW

File: views/dataset/refresh_publications.templ [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	"github.com/ugent-library/biblio-backoffice/ctx"
+	"github.com/ugent-library/biblio-backoffice/models"
+)
+
+templ RefreshPublications(c *ctx.Ctx, dataset *models.Dataset, relatedPublications []*models.Publication) {
+	<div hx-swap-oob="innerHTML:#modals"></div>
+	<div hx-swap-oob="innerHTML:#publications-body">
+		@publicationsBody(c, dataset, relatedPublications)
+	</div>
+}
```
Comment (verheyenkoen) [inline views/dataset/refresh_publications.templ:13]: Zie ReplaceModal templ view

File: views/dataset/show_nav.templ [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"github.com/ugent-library/biblio-backoffice/ctx"
+	"github.com/ugent-library/biblio-backoffice/models"
+	"github.com/ugent-library/biblio-backoffice/views"
+	"net/url"
+)
+
+type NavArgs struct {
+	ActiveSubNav string
+	SubNavs      []string
```
Comment (nics) [inline views/dataset/show_nav.templ:12]: I would just make this a package level var (see views/publication/sub_nav.templ)
ctx.Ctx now has a SubNav field that you can use, no need top pass it around, and there is a SetSubNav middleware

File: views/dataset/show_publications.templ [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package dataset
+
+import (
+	"github.com/ugent-library/biblio-backoffice/ctx"
+	"github.com/ugent-library/biblio-backoffice/models"
+)
+
+type NavPublicationsArgs struct {
```
Comment (nics) [inline views/dataset/show_publications.templ:8]: ctx contains the subnav, no need for an args struct

File: views/dataset/show_nav.templ (hunk 2 above)
Comment (nicolasfranck) [reply views/dataset/show_nav.templ:12]: Didn't know that!

File: views/dataset/refresh_publications.templ (hunk 1 above)
Comment (nicolasfranck) [reply views/dataset/refresh_publications.templ:13]: agreed but this doesn't work like that ReplaceModal..

File: views/dataset/show_nav.templ (hunk 2 above)
Comment (nics) [reply views/dataset/show_nav.templ:12]: fresh of the press ;-)

Comment (nicolasfranck) [reply views/dataset/show_nav.templ]: updated

File: views/dataset/suggest_publications.templ [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	</ul>
+}
+
+templ suggestPublicationSummaryActions(c *ctx.Ctx, dataset *models.Dataset, publication *models.Publication) {
+	<div class="bc-toolbar-item ms-auto ms-lg-0">
+		<div class="c-button-toolbar">
+			<button
+				class="btn btn-primary"
+				hx-post={ views.URL(c.PathTo("dataset_create_publication", "id", dataset.ID)).String() }
+				hx-headers={ fmt.Sprintf("{\"If-Match\": \"%s\"}", dataset.SnapshotID) }
```
Comment (nics) [inline views/dataset/suggest_publications.templ:28]: backticks make this much more readable

**Review by nics — APPROVED** [review]

**Pull request merged**